runtime/src/accounts_db.rs

```diff
@@ -4302,24 +4304,32 @@ impl AccountsDb {
         })
     }
 
-    /// 'accounts' are about to be appended to an ancient append vec. That ancient append vec may already have some accounts.
-    /// Unref each account in 'accounts' that already exists in 'existing_ancient_pubkeys'.
-    /// As a side effect, on exit, 'existing_ancient_pubkeys' will now contain all pubkeys in 'accounts'.
-    fn unref_accounts_already_in_storage(
-        &self,
-        accounts: &[(&Pubkey, &StoredAccountMeta<'_>, u64)],
+    fn get_keys_to_unref_ancient<'a>(
+        accounts: &'a [(&Pubkey, &StoredAccountMeta<'_>, u64)],
         existing_ancient_pubkeys: &mut HashSet<Pubkey>,
-    ) {
+    ) -> HashSet<&'a Pubkey> {
         let mut unref = HashSet::<&Pubkey>::default();
         // for each key that we're about to add that already exists in this storage, we need to unref. The account was in a different storage.
         // Now it is being put into an ancient storage again, but it is already there, so maintain max of 1 ref per storage in the accounts index.
         // The slot that currently references the account is going away, so unref to maintain # slots that reference the pubkey = refcount.
         accounts.iter().for_each(|(key, _, _)| {
             if !existing_ancient_pubkeys.insert(**key) {
                 // this key exists BOTH in 'accounts' and already in the ancient append vec, so we need to unref it
-                unref.insert(*key);
+                unref.insert(key);
```

Comment:
Wondering how we were able to drop the dereference compared to old code here. It seems like key should still be a &&Pubkey, since accounts hasn't really changed (added lifetime, but it's still a slice).